### PR TITLE
[CLD-6537] Cloud Annual Renewals - Banners Implementation

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
Adds banners throughout the webapp to support CTA's around Self Serve Cloud Annual Renewals. 

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-6537

#### Screenshots
The following screenshots are from the perspective of a system administrator only.
x = number of days until renewal

**60 >= x > 30**
image
image

**30 >= x > 7**
image
image

**7 >= x >= 0**
The actual subscription expiration time is on the 0th day but can be at any time during that day (it's exactly 1 year from the time they signed up, to the second). The below shows the banner they'd see if they had not opted to renew, and the subscription had yet to expire (for ex: subscription expires @ 1PM Dec 31, this is the view at 10AM Dec 31)
image
image

**0 >= x >= -14**
As mentioned above, if the subscription has expired and payment failed (due to either a CC decline, or lack of a CC) the below banners will be displayed. **NOTE:** These banners will be visible to BOTH end users and system administrators
Admin view:
image
image
non-admin view:
Note the lack of a CTA button, as the end user won't be able to pay. We could potentially place a contact sales button here, but that may result in someone unable to actually go through with the upgrade contacting sales. 
image

#### Release Note

```release-note
None
```

### CODE REVIEW

**Comment by nickmisasi** [comment]
@asaadmahmood Due to the nature of these banners being driven by data coming from Stripe, it's difficult (basically not really possible) to get these to appear in the spinwick normally, unless you know how to time travel :) 

If you want to drill down past the screenshots, I can put some time on our calendars to have you go through it on my local instance. 


Some things to note:
The CTA button styling on the banner is slightly different from designs - the one in my screenshots was the existing styling for that button across banners in the webapp.
The designs didn't have a blue banner in the system console subscription's page - it wasn't much effort to include it while I was making the red ones, so I went ahead and did it. I did not make any of the subscription page one's dismissable (as opposed to the system-wide blue one's that are) as I figured this would be a good persistent reminder to those visiting the system console. Lmk what you think. 
I switched up some of the copy from Figma to be a bit more complete, but will update with any suggestions from Carrie as she reviews.

**Review by cwarnermm — APPROVED** [review]
Please end all in-product sentences with a period. Applies to the admin message on the System Console > Subscription page, and the "has expired" banner CTAs. Thanks! Approving.

**Comment by nickmisasi** [comment]
Will do. Thanks @cwarnermm!

**Comment by johndavidlugtu** [comment]
@nickmisasi is the CTA for sales-serve customers "Contact sales"?

Or will these banners not appear at all?

**Comment by nickmisasi** [comment]
@johndavidlugtu These will not appear for sales serve customers

**Comment by asaadmahmood** [comment]
@nickmisasi I think looks good, is there a way to see what this would look like with some items?

Screenshot 2023-11-09 at 12 34 52 PM

**Review by asaadmahmood — APPROVED** [review]
Looks good I think.

**Comment by nickmisasi** [comment]
@asaadmahmood with items it would look like this:
image


It's only buggy like that in the PR screenshot because everything's mocked out locally

**Review by gabrieljackson — APPROVED** [review]

File: server/channels/api4/cloud.go [hunk 1]
```diff
@@ -93,20 +93,26 @@ func getSubscription(c *Context, w http.ResponseWriter, r *http.Request) {
 			ProductID:       subscription.ProductID,
 			IsFreeTrial:     subscription.IsFreeTrial,
 			TrialEndAt:      subscription.TrialEndAt,
+			EndAt:           subscription.EndAt,
+			CancelAt:        subscription.CancelAt,
+			DelinquentSince: subscription.DelinquentSince,
```
[the remark this answers is not in the document]
Comment (esarafianou) [reply server/channels/api4/cloud.go:98]: @nickmisasi Why should simple members get notified that the subscription has expired?

[the remark this answers is not in the document]
Comment (nickmisasi) [reply server/channels/api4/cloud.go:98]: @esarafianou 14 days after expiration, the workspace is deleted. During the Cloud Free/Monthly deprecation, we had a large number of users claim to have not seen the persistent notification banners informing them of the impending deletion - these banners were for admins only. They only found out when it was too late and the workspace was already deleted.

The banner will show to all users during the 14 days after expiration - that way they can get in touch with their workspace owners if they're still using the service and have it paid for.

**Review by M-ZubairAhmed — APPROVED** [review]

File: server/channels/api4/cloud.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (esarafianou) [reply server/channels/api4/cloud.go:98]: Thanks for explaining, I don't see any issues with exposing these considering the previous feedback and the benefits our customers will take.

**Pull request merged**